ssl/ssl_lib.cc

```diff
@@ -929,6 +931,24 @@ static int ssl_do_post_handshake(SSL *ssl, const SSLMessage &msg) {
   return 1;
 }
 
+// Sets pha_ext value in SSL_CTX
+void SSL_CTX_set_post_handshake_auth(SSL_CTX *ctx, int val) {
+  ctx->pha_ext = val;
+}
+
+// Sets pha_ext value in SSL
+void SSL_set_post_handshake_auth(SSL *ssl, int val) {
+  ssl->s3->pha_ext = val;
```

Comment:
Won't setting these flip the internal state of the post handshake authentication? 
This doesn't seem to be part of the function contract and the internal state of PHA shouldn't be exposed for the user to configure. We should also separate the configuration and the state management of PHA into different fields. OpenSSL also does this with two variables, one for enabling pha, one for maintaining the state.
https://github.com/openssl/openssl/blob/9c8d04dbec03172d6ffe4eaa38ea4b1ac2741f26/ssl/ssl_local.h#L1725-L1726C9

You can consider using a `bool` internally to maintain whether PHA is enabled or not. Something clear like `post_handshake_state` and `post_handshake_enabled` would be nice :)

Reply:
Good call, I didn't make that distinction. Have added in another field to manage that.